### ISSUE TITLE
Question about p_paginate  and mapping sheet name with sheet id

### ISSUE THREAD

**Issue by shenyenwu** [mbleron/ExcelGen#66]
Hi
After migrated 4.2 version into my DB，I got 2 errors while generating excel.
1. for example, when I call the following function, it returns no_data_found, it works only when p_paginate is false. 
sheetId := ExcelGen.addSheetFromCursor(
    p_ctxId     => ctxId
  , p_sheetName => 'sheet${PNUM}'
  , p_rc        => rc
  , p_tabColor  => 'DeepPink'
  , p_paginate  => true 
  );

or 
sheetId := ExcelGen.addSheetFromCursor(
    p_ctxId     => ctxId
  , p_sheetName => 'sheet${PNUM}'
  , p_rc        => rc
  , p_tabColor  => 'DeepPink'
  , p_paginate  => true 
, p_pageSize  => 10000
  );

2. If I use sheet name to add sheet, and call setHeader afterward, it seems like converts sheet name into upper case at the background and cause setHeader failed. (sheet name not matched); This situation can be avoid by using sheet_id all the way.

Wondering if I did something wrong.
Thank you for your great help.

**Comment by mbleron** [comment]
Hi,

For point 1,

You're not doing anything wrong, there's a flaw in the implementation logic when pagination is used.
The issue was introduced with formula support. Since paginated sheet name can be parameterized, that name is not resolved at the time the current sheet is passed to the formula context (even though there's no formula involved), hence NO_DATA_FOUND exception.

As a temporary hot fix, we may alter the following line by replacing:
https://github.com/mbleron/ExcelGen/blob/2d32ded190de05aa34ac6ff3a137dc79b38aadb6/plsql/ExcelGen.pkb#L4600
with
```
    if not sd.pageable then
      ExcelFmla.setCurrentSheet(sd.sheetName);
    end if;
```
The side effect is that we won't be able to use scoped cell references, or names in paginated sheets.

In the meantime, I'll need to rethink all this a bit...

**Comment by mbleron** [comment]
Point 2.

Yes, another bug there. The sheetName should be upper-ized before accessing sheetIndexMap:
https://github.com/mbleron/ExcelGen/blob/2d32ded190de05aa34ac6ff3a137dc79b38aadb6/plsql/ExcelGen.pkb#L6051

But as you noticed, sheetId is the way to go now, as overloads using sheetName are all deprecated.

**Comment by mbleron** [comment]
Fixed in v4.2.1.
